src/gsheets_read.cpp

```diff
@@ -149,8 +151,18 @@ unique_ptr<FunctionData> ReadSheetBind(ClientContext &context, TableFunctionBind
             use_explicit_sheet_name = true;
             sheet_name = kv.second.GetValue<string>();
 
+            // Try to extract range from sheet if user provided a name using A1 notation (e.g. `Sheet1!A1:B7`)
+            // TODO: handle scenarios where sheet name may contain `!` when not using A1 notation
```

Comment:
This is how google sheets escapes ! in sheet names

![CleanShot 2025-02-04 at 13 31 10@2x](https://github.com/user-attachments/assets/4fb0b929-160a-4b40-91eb-c6b307e8401a)

Reply:
Want me to go ahead and add this fix to this PR?

Reply:
Im not 100% clear how we would do it - ignore any `!` inside quotes? eg `'!'`?

Reply:
I'm also happy to leave this edge case in for now, feels like a low percentage use case

Reply:
I'm trying a few things out. If it becomes something that would make the overall range feature more brittle then I'll leave it alone. I think having it would be nice especially for copy-pasted URLs